src/utils/dom.ts

```diff
@@ -20,6 +20,66 @@ export function getElementProp(el: Element, prop: string, fallbackValue: any, cr
   return closest ? closest.getAttribute(prop) : fallbackValue;
 }
 
+function getRootNode(el: Element): HTMLDocument | ShadowRoot {
+  return el.getRootNode() as HTMLDocument | ShadowRoot;
+}
+
+export function queryElementsCrossShadowBoundary<T extends Element = Element>(
```

Comment:
@jcfranco I'm not sure if having separate functions for these dom queries is good or if we want something like `getSlotted` that allows `all`?

We need to be able to do the following queries:

- querySelector
- querySelectorAll
- closest
- Potentially `getElementById` if we feel the slight performance increase is worth it over using `querySelector(#id)`

All of these will query the shadowDom first then go up the chain using `getRootNode()`

For `querySelectorAll`, we may want an option to just return the first set of results or return all of the results from each shadowDom and document.

Reply:
Let me know what you are thinking about this...

Its rather complicated so I'm not sure about brining it up in a sync meeting.

Reply:
Sorry, I missed this comment. I like the idea of having a single util to would allow us to call DOM APIs across shadow boundary. Maybe something like (very rough):

```ts
domQuery<T extends HTMLElement>(
  method, // `selector`, `closest`, others?
  options // `all`, others?
): Element | Elements | null; // we can fine tune results based on input arguments
```

We can iterate on this if you think it'd be useful.

Reply:
Yeah let's iterate on it.  I'd like to get this in so that the issue is fixed.

Since this is all internal, lets proceed with the current name and we can revisit with a complete dom query API.

Reply:
Sounds good.

Reply:
Cool i'll create a new issue after merge.